department-of-veterans-affairs/caseflow
# CAVC Remand: Litigation Support Form (Front End)

Resolves #14795. Resolves CASEFLOW-65

### Description
Creates a new Post-dispatch actions area on the Case Details page which houses the Add Cavc Remand option.  This option is only available to member of the `CavcLitigationSupport` org and is also hidden behind the `cavc_remand` feature toggle.

Also creates the new Cavc Intake Form, which in conjunction with the work completed in #15351 and #15469 will allow Cavc litigation users to create a new Cavc appeal stream.

### Acceptance Criteria
- [x] "+Add Cavc remand" button is behind `cavc_remand` feature toggle
- [x] Only Cavc Litigation Support users can see the button
- [ ] Start CAVC Remand Documentation to describe what cavc remand features caseflow supports (to be updated as more cavc functionality is added)
- [x] Add CAVC Court Order Button to Case Details page of a Dispatched/Completed appeal that presents the form in a new top section
- [x] Cavc Litigation Support user can add the following required fields:
  - [ ] court docket number
  - [ ] Attorney representation
  - [ ] Cavc judge
  - [ ] Decision type (remand, death dismissal, etc)
  - [ ] type of remand (JMR, JMPR) on the form (only if remand)
  - [ ] Court decision date
  - [ ] Judgement date
  - [ ] Mandate date
  - [ ] issues required to be decided on in the new appeal stream
- [x] Confirmation alert appears with the following copy when the form is completed & submitted: 
        "You have successfully created a CAVC Remand case
         This has generated a Send 90 Day Letter task in your team's queue"
- [ ] Ensure users can copy and paste values from the cavc website without modifying data (see [here](https://github.com/department-of-veterans-affairs/caseflow/issues/14795#issuecomment-722592391) for how to get to the publicly available records)

### Testing Plan
1. Enable cavc_remand feature toggle
1. Find a dispatched appeal (search for vet 500000000)
1. Click "+ Add CAVC Remand"
1. Press submit and confirm validations
1. Confirm selecting "Death Dismissal" hides remand subtype
1. Fill out the form, keeping JMR selected
1. Deselect an issue
1. Press submit and confirm error message
1. Select all issues and press submit.
1. Confirm success message on case details of new appeal
1. Confirm case details shows correct information entered

### User Facing Changes

Workflow|UI
 ---|---
New button|<img width="1447" alt="Screen Shot 2020-11-17 at 4 29 21 PM" src="https://user-images.githubusercontent.com/45575454/99457467-07565c00-28f9-11eb-8185-a590650e414d.png">
Validations|<img width="917" alt="Screen Shot 2020-11-17 at 1 57 50 PM" src="https://user-images.githubusercontent.com/45575454/99457433-f60d4f80-28f8-11eb-9250-3f7ecd7dc8ea.png"><img width="915" alt="Screen Shot 2020-11-17 at 1 57 58 PM" src="https://user-images.githubusercontent.com/45575454/99457438-f86fa980-28f8-11eb-9a2c-fa2ebe3b2bb0.png">
Back end error|<img width="818" alt="Screen Shot 2020-11-17 at 3 54 27 PM" src="https://user-images.githubusercontent.com/45575454/99457452-00c7e480-28f9-11eb-9489-87eefd9796a6.png">
Success Alert and new appeal|<img width="1115" alt="Screen Shot 2020-11-17 at 5 21 14 PM" src="https://user-images.githubusercontent.com/45575454/99457662-53090580-28f9-11eb-89fa-02b65183df65.png">

### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*
